### DISCUSSION TITLE
How to create a custom layer or find CustomLayerHost from mapbox ?

### DISCUSSION THREAD

**Discussion by D1maD1m0nd** [maplibre/maplibre-native#2359] [Q&A]
Hello. I want to create a custom layer for icons so that I can set the font and render, change the height, etc.
I found the examples below on the Internet, but I did not find the CustomLayerHost interface from the example
How can I make a similar example on Map libr (without C++)
**version = 10.2.0**

[Links removed -- @louwers]

**Comment by louwers** [comment]
Please explain what you are trying to achieve instead of linking to proprietary code. We're very cautious not to look at proprietary code to not inadvertently violate copyright of Mapbox. I removed links to proprietary code in a Mapbox repo, thanks for understanding!

That said, you can find an example custom layer for Android here: https://github.com/maplibre/maplibre-native/blob/cd6a0ab11fcd9b5bbfee87896a14159d0ed56256/platform/android/src/example_custom_layer.cpp#L126

**Reply by D1maD1m0nd** [reply]
Is it possible to make a custom layer in Java on openGl ES 20?
My task is to create a new layer to place icons on it and control their rendering, location and attributes

**Reply by louwers** [reply]
Yes this is possible, but you will have to use the 10.3.x MapLibre Native Android releases, we will upgrade to OpenGL ES 3.0 in 11.0.0.

**Reply by D1maD1m0nd** [reply]
You can show an example of creating a custom layer in Java and OpenGL ES for the Maplibre android SDK ('org.maplibre.gl:android sdk:10.3.0'). There is already an example on the site, but it uses some CustomLayerHost that I haven't found anywhere to use.
Or you can write an algorithm on how I should proceed to develop my own layer in OpenGL ES with Maplibre, I don’t quite understand where to start

**Reply by louwers** [reply]
Actually, a custom layer needs to be implemented using C++ at the moment.

But I doubt using a custom layer is needed for your use case. You certainly do not need it for icons. Can you share a bit more information?

**Reply by D1maD1m0nd** [reply]
I am working on a project where we place 3d models of a building on a map using polygons in the FillExtrusionLayer layer. There is a problem with placing icons above rooms in a building that are on high floors, because icons can only be placed on the plane of the map itself and they cannot be set height relative to ground level in SymbolLayer. And that's why I wanted to use a custom layer to solve this problem.
